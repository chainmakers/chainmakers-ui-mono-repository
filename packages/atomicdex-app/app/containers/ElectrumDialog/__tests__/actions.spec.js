import {
  showElectrumDialog,
  hideElectrumDialog,
  addElectrum,
  removeElectrum
} from '../actions';
import {
  ELECTRUM_DIALOG_SHOW,
  ELECTRUM_DIALOG_HIDE,
  ELECTRUM_ADD,
  ELECTRUM_REMOVE
} from '../constants';

describe('containers/ElectrumDialog/actions/showElectrumDialog', () => {
  it('should showElectrumDialog should create showElectrumDialog action', () => {
    expect(showElectrumDialog()).toMatchSnapshot();
  });

  it('should return the correct type and the passed name', () => {
    const expectedResult = {
      type: ELECTRUM_DIALOG_SHOW
    };

    expect(showElectrumDialog()).toEqual(expectedResult);
  });
});

describe('containers/ElectrumDialog/actions/hideElectrumDialog', () => {
  it('should hideElectrumDialog should create hideElectrumDialog action', () => {
    expect(hideElectrumDialog()).toMatchSnapshot();
  });

  it('should return the correct type and the passed name', () => {
    const expectedResult = {
      type: ELECTRUM_DIALOG_HIDE
    };

    expect(hideElectrumDialog()).toEqual(expectedResult);
  });
});

describe('containers/ElectrumDialog/actions/addElectrum', () => {
  const payload = {
    coins: ['KMD', 'BTC']
  };
  it('should addElectrum should create addElectrum action', () => {
    expect(addElectrum(payload)).toMatchSnapshot();
  });

  it('should return the correct type and the passed name', () => {
    const expectedResult = {
      type: ELECTRUM_ADD,
      payload
    };

    expect(addElectrum(payload)).toEqual(expectedResult);
  });
});

describe('containers/ElectrumDialog/actions/removeElectrum', () => {
  const payload = {
    coins: ['KMD', 'BTC']
  };
  it('should removeElectrum should create removeElectrum action', () => {
    expect(removeElectrum(payload)).toMatchSnapshot();
  });

  it('should return the correct type and the passed name', () => {
    const expectedResult = {
      type: ELECTRUM_REMOVE,
      payload
    };

    expect(removeElectrum(payload)).toEqual(expectedResult);
  });
});
