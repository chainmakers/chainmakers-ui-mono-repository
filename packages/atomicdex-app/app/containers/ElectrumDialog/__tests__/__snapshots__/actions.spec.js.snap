// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/ElectrumDialog/actions/addElectrum should addElectrum should create addElectrum action 1`] = `
Object {
  "payload": Object {
    "coins": Array [
      "KMD",
      "BTC",
    ],
  },
  "type": "atomicapp/ElectrumDialog/ELECTRUM_ADD",
}
`;

exports[`containers/ElectrumDialog/actions/hideElectrumDialog should hideElectrumDialog should create hideElectrumDialog action 1`] = `
Object {
  "type": "atomicapp/ElectrumDialog/ELECTRUM_DIALOG_HIDE",
}
`;

exports[`containers/ElectrumDialog/actions/removeElectrum should removeElectrum should create removeElectrum action 1`] = `
Object {
  "payload": Object {
    "coins": Array [
      "KMD",
      "BTC",
    ],
  },
  "type": "atomicapp/ElectrumDialog/ELECTRUM_REMOVE",
}
`;

exports[`containers/ElectrumDialog/actions/showElectrumDialog should showElectrumDialog should create showElectrumDialog action 1`] = `
Object {
  "type": "atomicapp/ElectrumDialog/ELECTRUM_DIALOG_SHOW",
}
`;
